compiler/dialects/tsl.py

```diff
@@ -53,59 +53,58 @@ def get_bound_ops(
             and depth. This is used to keep track of which sequence of operations
             was made for which TSL Stride.
         """
-        if isinstance(memref_op_or_shapes, SSAValue | Operation):
-            memref = memref_op_or_shapes
-            shapes = None
-        else:
-            shapes = memref_op_or_shapes
-            memref = None
 
         result: list[Operation] = []
         result_mapping: dict[(int, int), Operation] = {}
 
         tsl = self.data
 
+        # if the argument passed is a memref, generate shape operation
+        # list by using the dim operation
+        if isinstance(memref_op_or_shapes, SSAValue | Operation):
+            memref = memref_op_or_shapes
+            shapes = []
+            for dim in range(tsl.dimension()):
+                dim_index_op = Constant.from_int_and_width(dim, IndexType())
+                dim_op = Dim.from_source_and_index(memref, dim_index_op)
+                result.extend([dim_index_op, dim_op])
+                shapes.append(dim_op)
+        else:
+            shapes = memref_op_or_shapes
+
```

Comment:
The strategy in this code is:
if the shape ops have been supplied, use the shape ops
if the memref ops have been supplied, create dim ops to use as shape ops

This was already in the code but is now refactored to be nicely in one place